prow/cmd/build/controller.go

```diff
@@ -57,6 +58,20 @@ const (
 	controllerName = "prow-build-crd"
 )
 
+type pjClient struct {
+	pjc       prowjobset.Interface
+	namespace string
+}
+
+func (c *pjClient) ReplaceProwJob(name string, pj prowjobv1.ProwJob) (prowjobv1.ProwJob, error) {
+	updatedPj, err := c.pjc.ProwV1().ProwJobs(c.namespace).Update(&pj)
+	if updatedPj != nil {
```

Comment:
IIRC these methods don't return `nil` but empty structs, but this is techcnially correct so meh

Reply:
Comparing a value to nil will return a compiler error. update returns a pointer.

Reply:
A pointer to an empty struct, not a nil pointer is what I meant to say

Reply:
The signature was manly driven by the existing plank client: https://github.com/kubernetes/test-infra/blob/bbad344943041822cfe15d2840bc64da62164550/prow/plank/controller.go#L47